### PR TITLE
Add instructions on how to add a new cop.

### PR DESCRIPTION
**Pull request by blainesch**
These were things I ran into while I was creating my first cop that I believe will help onboard others.

There's one thing I left blank which is `VariableForce`. I'm not entirely sure, but I believe this does special things , so instead of getting a `node` you'd get a few nodes that match a type of group? For instance all the nodes in an assignment. Does anybody want to point me in the right direction here?

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -22,6 +22,41 @@ $ rubocop -V
...
+1) Create your cop in `./lib/rubocop/cop/{type}/{name}.rb`
+2) Require your cop in `./lib/rubocop.rb`
+3) Your test should be created at `./spec/rubocop/cop/{type}/{name}_spec.rb`
+4) Add your rule to either the `config/disabled.yml` or `config/enabled.yml` file depending on if it should be enabled by default or not. Cops that are not based on style guide rules should be disabled (normally) by default.
+
+### Finding your offenses
+
+#### Investigate
+
+The [investigate method](lib/rubocop/cop/style/semicolon.rb#L11) gets passed a [processed source](lib/rubocop/processed_source.rb). This object can be asked questions about the source, retrieve all of the source tokens and get specific lines as strings.
```
Comment (vassilevsky) [inline CONTRIBUTING.md]: If `lib/rubocop/cop/style/semicolon.rb` changes in the future, the link will become irrelevant.
One way to solve this is to [bind to the current SHA](http://andrew.yurisich.com/work/2014/07/16/dont-link-that-line-number/).

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -22,6 +22,41 @@ $ rubocop -V
 
 * Include any relevant code to the issue summary.
 
+## Creating Cops
+
+### Create the cop files
+
+1) Create your cop in `./lib/rubocop/cop/{type}/{name}.rb`
```
Comment (bquorning) [inline CONTRIBUTING.md]: The syntax for Markdown lists is `1.` instead of `1)`.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -22,6 +22,41 @@ $ rubocop -V
...
+Yeah, I got nothing. What's this do?
+
+### Auto Correct
+
+Rubocop allows you to auto correct code based on your rules. You add a lambda to an array of `corrections`, which then get processed. Your lambda will be passed a [corrector](lib/rubocop/cop/corrector.rb) instance, and has access to the node that was passed into the main `autocorrect` method.
+
+### Tests
+
+~~~ bash
+$ rake && rubocop
```
Comment (Koronen) [inline CONTRIBUTING.md]: Just
```sh
$ rake
```
should be enough here. The default Rake task [includes running RuboCop over itself](https://github.com/Koronen/rubocop/blob/dd2302039e2667f9d4cd95dc4f24548af06c719f/Rakefile#L25-28).

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -22,6 +22,41 @@ $ rubocop -V
...
+3) Your test should be created at `./spec/rubocop/cop/{type}/{name}_spec.rb`
+4) Add your rule to either the `config/disabled.yml` or `config/enabled.yml` file depending on if it should be enabled by default or not. Cops that are not based on style guide rules should be disabled (normally) by default.
+
+### Finding your offenses
+
+#### Investigate
+
+The [investigate method](lib/rubocop/cop/style/semicolon.rb#L11) gets passed a [processed source](lib/rubocop/processed_source.rb). This object can be asked questions about the source, retrieve all of the source tokens and get specific lines as strings.
+
+#### On specific nodes
```
Comment (bbatsov) [inline CONTRIBUTING.md]: On specific nodes -> Processing source nodes

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -22,6 +22,41 @@ $ rubocop -V
...
+
+Special methods get triggered, if they exist, for individual nodes. You can target a specific node to get to a single method using this. Defining a method `on_class`, for example, would get triggered everytime a class node is found in the source, and that node is passed in.
```
Comment (bbatsov) [inline CONTRIBUTING.md]: Special methods -> Callback methods.

Comment (bbatsov) [inline CONTRIBUTING.md]: Special -> callback

The second sentence here is kind of confusing.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -22,6 +22,41 @@ $ rubocop -V
...
+
+A [list of node types](https://github.com/whitequark/parser/blob/master/lib/parser/meta.rb) are available at the [parser](https://github.com/whitequark/parser) repo.
+
+#### Variable Force
```
Comment (bbatsov) [inline CONTRIBUTING.md]: You can remove this section for now.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -22,6 +22,41 @@ $ rubocop -V
...
+
+Yeah, I got nothing. What's this do?
+
+### Auto Correct
+
+Rubocop allows you to auto correct code based on your rules. You add a lambda to an array of `corrections`, which then get processed. Your lambda will be passed a [corrector](lib/rubocop/cop/corrector.rb) instance, and has access to the node that was passed into the main `autocorrect` method.
```
Comment (bbatsov) [inline CONTRIBUTING.md]: RuboCop

**Comment by bbatsov** [comment]
@blainesch Do the suggested updates and I'll merge this. Guess me & @jonas054 will improve/extend it afterwards.

**Comment by bbatsov** [comment]
A related note - probably this should be part of the README, not CONTRIBUTING.

**Comment by bbatsov** [comment]
Closind due to lack of activity.

**Pull request closed without merging**